flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -98,6 +101,14 @@ public PartitionOperator(DataSet<T> input, Keys<T> pKeys, Partitioner<?> customP
 		this.customPartitioner = customPartitioner;
 		this.distribution = distribution;
 	}
+
+	public PartitionOperator<T> withOrders(Order... orders) {
```

Comment:
Hi. I started working on this change, but I don't quite know how should I treat keyExpression (with wildcards especially). 

Lets take some complex example:

```
TypeInformation<Tuple3<Integer, Pojo1, PojoWithMultiplePojos>> ti =
		new TupleTypeInfo<>(
			BasicTypeInfo.INT_TYPE_INFO,
			TypeExtractor.getForClass(Pojo1.class),
			TypeExtractor.getForClass(PojoWithMultiplePojos.class)
		);

ek = new ExpressionKeys<>(new String[] {"f2.p1.*", "f0"}, ti);

public static class Pojo1 {
	public String a;
	public String b;
}
public static class Pojo2 {
	public String a2;
	public String b2;
}
public static class PojoWithMultiplePojos {
	public Pojo1 p1;
	public Pojo2 p2;
	public Integer i0;
}
```

What should be the output of `ek.getOriginalKeyFieldTypes`?